src/gui/transferlistdelegate.cpp

```diff
@@ -53,168 +47,36 @@ TransferListDelegate::TransferListDelegate(QObject *parent)
-    case TransferListModel::TR_DLLIMIT: {
-            const qlonglong limit = index.data().toLongLong();
-            if (hideValues && !limit)
-                break;
-            opt.displayAlignment = Qt::AlignRight | Qt::AlignVCenter;
-            QItemDelegate::drawDisplay(painter, opt, opt.rect, limit > 0 ? Utils::Misc::friendlyUnit(limit, true) : QString::fromUtf8(C_INFINITY));
-        }
-        break;
-    case TransferListModel::TR_TIME_ELAPSED: {
-            const qlonglong elapsedTime = index.data().toLongLong();
-            const qlonglong seedingTime = index.data(Qt::UserRole).toLongLong();
-            const QString txt = (seedingTime > 0)
-                ? tr("%1 (seeded for %2)", "e.g. 4m39s (seeded for 3m10s)")
-                    .arg(Utils::Misc::userFriendlyDuration(elapsedTime)
-                        , Utils::Misc::userFriendlyDuration(seedingTime))
-                : Utils::Misc::userFriendlyDuration(elapsedTime);
-            QItemDelegate::drawDisplay(painter, opt, opt.rect, txt);
-        }
-        break;
-    case TransferListModel::TR_ADD_DATE:
-    case TransferListModel::TR_SEED_DATE:
-        QItemDelegate::drawDisplay(painter, opt, opt.rect, index.data().toDateTime().toLocalTime().toString(Qt::DefaultLocaleShortDate));
-        break;
-    case TransferListModel::TR_RATIO_LIMIT:
-    case TransferListModel::TR_RATIO: {
-            const qreal ratio = index.data().toDouble();
-            if (hideValues && (ratio <= 0))
-                break;
-            QString str = ((ratio == -1) || (ratio > BitTorrent::TorrentHandle::MAX_RATIO)) ? QString::fromUtf8(C_INFINITY) : Utils::String::fromDouble(ratio, 2);
-            opt.displayAlignment = Qt::AlignRight | Qt::AlignVCenter;
-            QItemDelegate::drawDisplay(painter, opt, opt.rect, str);
-        }
-        break;
-    case TransferListModel::TR_QUEUE_POSITION: {
-            const int queuePos = index.data().toInt();
-            opt.displayAlignment = Qt::AlignRight | Qt::AlignVCenter;
-            if (queuePos > 0)
-                QItemDelegate::paint(painter, opt, index);
-            else
-                QItemDelegate::drawDisplay(painter, opt, opt.rect, "*");
-        }
-        break;
-    case TransferListModel::TR_PROGRESS: {
-            const qreal progress = index.data().toDouble() * 100;
+    const qreal progress = index.data(Qt::UserRole).toReal();
 
-            QStyleOptionProgressBar newopt;
-            newopt.rect = opt.rect;
-            newopt.text = ((progress == 100)
-                ? QString("100%")
-                : (Utils::String::fromDouble(progress, 1) + '%'));
-            newopt.progress = static_cast<int>(progress);
-            newopt.maximum = 100;
-            newopt.minimum = 0;
-            newopt.state |= QStyle::State_Enabled;
-            newopt.textVisible = true;
+    QStyleOptionProgressBar newopt;
+    newopt.rect = opt.rect;
+    newopt.text = ((static_cast<int>(progress) == 100)
```

Comment:
I think this can simplified to `newopt.text = index.data().toString()`

Reply:
👍 
Fixed.